### ISSUE TITLE
Bug: Invisible header if list.size() < 2

### ISSUE THREAD

**Issue by fnk0** [blipinsk/RecyclerViewHeader#46]
I'm getting a weird problem when the size of the list is 0 or 1. The header space is there, but the header itself is invisible. 

[Example multiple items in the list](http://i67.tinypic.com/15wcn10.png)

[Example with 1 item in the list](http://i64.tinypic.com/157f88i.png)

**Comment by blipinsk** [comment]
I will try to look into that later today

**Comment by fnk0** [comment]
If it helps it only happens in the new version. I was using 1.x before and it worked fine for lists with 0 or 1 items before. 

Also it happens for LinearLayoutManager and GridLayoutManager.
Inspecting the ViewHiearchy shows that the View is in there with a height and Visibility is VISIBLE. But is not being draw :(

Thanks for the work on this library!: )

**Comment by fnk0** [comment]
Update: I have a pull to refresh in this page and after I pull to refresh the item header suddenly appears after the refresh... it's very weird specially since it is the same data!

**Comment by KrzysztofWrobel** [comment]
I experience the same issue. Any update on this one @blipinsk?

**Comment by KrzysztofWrobel** [comment]
I've found the place:

```
private void onScrollChanged() {
        hidden = !layoutManager.isFirstRowVisible();
        RecyclerViewHeader.super.setVisibility(hidden ? INVISIBLE : intendedVisibility);
        if (!hidden) {
            final int translation = calculateTranslation();
            if (isVertical) {
                setTranslationY(translation);
            } else {
                setTranslationX(translation);
            }
        }
    }

```
When there are no items "hidden" is set to "True" and header's visibility is set to INVISIBLE. 
I will try to fork and see if I can fix this issue.

**Comment by blipinsk** [comment]
Hi guys!
Sorry for not much of a response earlier, I had a lot of work to do that I wasn't really expecting to have. 

I have looked at the issue and indeed there is a problem with the `hidden = !layoutManager.isFirstRowVisible();` inside the `onScrollChanged` method. I didn't post a fix because I wasn't sure about the exact conditions for the issue to appear. I didn't want to post a partial solution to the problem. @fnk0 wrote that it applies to situations in which the list is size `0` or `1` but it doesn't happen for me if the size is `1`... are there any special conditions?

**Comment by fnk0** [comment]
I don't think so as far as I can tell only happens if adapter.count < 2. 
Right now I have a very hacky fix on which I add a empty element to my list if the size of the list is 1 (there's no point in having a header with lists of size zero for me) and set a flag on my adapter to set the visibility of that 2nd item to GONE. 

This was fairly easy for me to implement since I had a BaseAdapter for all my adapters so it didn't required much work. 

The only downside of that is that I get a extra line separator at the end of that element.

**Comment by blipinsk** [comment]
So it happens for you if the `adapter.count< 2` ? So if the `adapter.count==1`? It happens for me **only** if the `adapter.count==0`

**Comment by fnk0** [comment]
For me it happened when the list had 1 item. I think the link for the screenshot is gone now as tinypic removed it. But it happened for 1 item in the list as well.

**Comment by KrzysztofWrobel** [comment]
In my case it also happens only if the adapter.count==0.

Were able to fix it in my fork, but now when you know how to fix it. I believe there is no need to submit PR. :)

**Comment by niray** [comment]
It's my solve way :`  recyclerView.setAdatper(xxx);  
   recyclerView.scrollBy(0,1);` It's worked,because the recycleView content is not full  of screen.

**Comment by niray** [comment]
I find the condition : if the recycleView content can't filling of the screen, the header will display.invisible..

**Comment by blipinsk** [comment]
This bug has been fixed as a part of Release 2.0.1

Thanks for your contributions guys! 👍

**Comment by hanilozmen** [comment]
Hi,
I still face with the same problem with v2.0.1 . When I have 1 item in list, header is disappearing. Other than adding empty views, is there any workaround ? Here is my layout:

`<FrameLayout
                android:layout_width="match_parent"
                android:layout_height="wrap_content">

                <android.support.v7.widget.RecyclerView
                    android:id="@+id/areasRv"
                    android:layout_width="match_parent"
                    android:layout_height="wrap_content"
                    android:layout_gravity="center_horizontal|top"
                    android:focusable="false"
                    android:isScrollContainer="false"
                    android:overScrollMode="never" />

                <com.bartoszlipinski.recyclerviewheader2.RecyclerViewHeader
                    android:id="@+id/header"
                    android:layout_width="match_parent"
                    android:layout_height="wrap_content"
                    android:layout_gravity="center_horizontal|top">

                    <EditText
                              ......
                        />

                </com.bartoszlipinski.recyclerviewheader2.RecyclerViewHeader>
            </FrameLayout>`
